src/app/address/controllers/address/nonUKAddress.js

```diff
@@ -0,0 +1,124 @@
+const BaseController = require("hmpo-form-wizard").Controller;
+const countryList = require("../../data/countries.json");
+const {
+  validateBuildingAddress,
+} = require("../../validators/nonUKAddressValidator");
+
+class NonUKAddressController extends BaseController {
+  getValues(req, res, callback) {
+    super.getValues(req, res, (err, values) => {
+      values.addressCountryName = this.mapToName(
+        req.translate,
+        req.sessionModel.get("country")
+      );
+
+      Object.entries(req?.form?.errors || {})?.map(
+        ([fieldName, validationAttribute]) =>
+          (values[`${fieldName}InValid`] = this.getErrorMessage(
+            req.translate,
+            validationAttribute.key,
+            validationAttribute.type
+          ))
+      );
+
+      values.buildAddressErrorMessage = this.hasBuildingEmptyErrors(req) && {
```

Comment:
Would you mind adding a comment to this about why we needed to do it?

Reply:
The functionality is described [here](https://github.com/govuk-one-login/ipv-cri-address-front/pull/1008/commits/69b82946b89bd84e29881637ca5e4eeba3545647) i have added a comment

Reply:
hmm that commit doesn't appear to exist any more? 

Nice, yeah I can see from the code buildAddress error message only when all input fields are empty but could you please add a bit about the _why_? I think that's what might catch people out in the future. It's to do with the form focus, right?